Fix warning when using nullable reference types with OfType operator

#### Bug

Despite our best efforts, bugs can slip into releases or corner cases forgotten about. We will try our best to remedy the situation
and/or provide workarounds. Note that certain (odd) behaviors are by design and as such are not considered bugs.

> Which library version?
v5.0

> What are the platform(s), environment(s) and related component version(s)?
.net5.0

> What is the use case or problem?
I want to use the OfType operator for filtering out nulls like so:
```
Observable
    .Return<string?>("blah")
    .OfType<string>()
    .SelectMany(this.audioService.PlayAudioPair)
    .Subscribe();
```

> What is the expected outcome?
No warnings

> What is the actual outcome?
When compiled has a warning:
`warning CS8620: Argument of type 'IObservable<string?>' cannot be used for parameter 'source' of type 'IObservable<object>' in 'IObservable<string> Observable.OfType<string>(IObservable<object> source)' due to differences in the nullability of reference types.`

--- a/Ix.NET/Source/System.Linq.Async.Tests/System/Linq/Operators/OfType.cs
+++ b/Ix.NET/Source/System.Linq.Async.Tests/System/Linq/Operators/OfType.cs
@@ -1,6 +1,6 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT License.
-// See the LICENSE file in the project root for more information. 
+// See the LICENSE file in the project root for more information.
 
 using System;
 using System.Linq;
@@ -37,6 +37,18 @@ namespace Tests
 
             var e = ys.GetAsyncEnumerator();
             await HasNextAsync(e, "bar");
+            await HasNextAsync(e, "foo");
+            await NoNextAsync(e);
+        }
+
+        [Fact]
+        public async Task OfType_NotNullObject()
+        {
+            var xs = new object?[] { 42, null, "foo", null }.ToAsyncEnumerable();
+            var ys = xs.OfType<object>();
+
+            var e = ys.GetAsyncEnumerator();
+            await HasNextAsync(e, 42);
             await HasNextAsync(e, "foo");
             await NoNextAsync(e);
         }

--- a/Ix.NET/Source/System.Linq.Async/System/Linq/Operators/OfType.cs
+++ b/Ix.NET/Source/System.Linq.Async/System/Linq/Operators/OfType.cs
@@ -1,6 +1,6 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT License.
-// See the LICENSE file in the project root for more information. 
+// See the LICENSE file in the project root for more information.
 
 using System.Collections.Generic;
 using System.Threading;
@@ -25,14 +25,14 @@ namespace System.Linq
         /// <param name="source">The async-enumerable sequence that contains the elements to be filtered.</param>
         /// <returns>An async-enumerable sequence that contains elements from the input sequence of type TResult.</returns>
         /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
-        public static IAsyncEnumerable<TResult> OfType<TResult>(this IAsyncEnumerable<object> source)
+        public static IAsyncEnumerable<TResult> OfType<TResult>(this IAsyncEnumerable<object?> source)
         {
             if (source == null)
                 throw Error.ArgumentNull(nameof(source));
 
             return Core(source);
 
-            static async IAsyncEnumerable<TResult> Core(IAsyncEnumerable<object> source, [System.Runtime.CompilerServices.EnumeratorCancellation] CancellationToken cancellationToken = default)
+            static async IAsyncEnumerable<TResult> Core(IAsyncEnumerable<object?> source, [System.Runtime.CompilerServices.EnumeratorCancellation] CancellationToken cancellationToken = default)
             {
                 await foreach (var obj in source.WithCancellation(cancellationToken).ConfigureAwait(false))
                 {

